dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/gist/GistBuilder.java

```diff
@@ -0,0 +1,802 @@
+     */
+
+    public void addFetchParameters( BiConsumer<String, Object> dest,
+        BiFunction<String, Class<?>, Object> argumentParser )
+    {
+        for ( Field field : query.getFields() )
+        {
+            if ( field.getTransformationArgument() != null && field.getTransformation() != Transform.PLUCK )
+            {
+                dest.accept( "p_" + field.getPropertyPath(), field.getTransformationArgument() );
+            }
+        }
+        addCountParameters( dest, argumentParser );
+    }
+
+    public void addCountParameters( BiConsumer<String, Object> dest,
+        BiFunction<String, Class<?>, Object> argumentParser )
+    {
+        Owner owner = query.getOwner();
+        if ( owner != null )
+        {
+            dest.accept( "OwnerId", owner.getId() );
+        }
+        int i = 0;
+        for ( Filter filter : query.getFilters() )
+        {
+            Comparison operator = filter.getOperator();
+            if ( !operator.isUnary() )
+            {
+                Property property = context.resolveMandatory( filter.getPropertyPath() );
+                Object value = getParameterValue( property, filter, argumentParser );
+                dest.accept( "f_" + (i++), operator.isStringCompare()
+                    ? completeLikeExpression( operator, (String) value )
+                    : value );
+            }
+        }
+    }
+
+    private Object getParameterValue( Property property, Filter filter,
+        BiFunction<String, Class<?>, Object> argumentParser )
+    {
+        String[] value = filter.getValue();
+        if ( value.length == 0 )
+        {
+            return "";
+        }
+        if ( value.length == 1 )
+        {
+            return getParameterValue( property, filter, value[0], argumentParser );
+        }
+        return stream( value ).map( e -> getParameterValue( property, filter, e, argumentParser ) ).collect( toList() );
+    }
+
+    private Object getParameterValue( Property property, Filter filter, String value,
+        BiFunction<String, Class<?>, Object> argumentParser )
+    {
+        if ( value == null || property.getKlass() == String.class )
+        {
+            return value;
+        }
+        if ( isCollectionSizeFilter( filter, property ) )
+        {
+            return argumentParser.apply( value, Integer.class );
+        }
+        String valueAsJson = value;
+        Class<?> itemType = getBaseType( property );
+        if ( !(Number.class.isAssignableFrom( itemType ) || itemType == Boolean.class || itemType == boolean.class) )
+        {
+            valueAsJson = '"' + value + '"';
+        }
+        return argumentParser.apply( valueAsJson, itemType );
+    }
+
+    private static Object completeLikeExpression( Comparison operator, String value )
+    {
+        switch ( operator )
+        {
+        case LIKE:
+        case NOT_LIKE:
+            return value != null && (value.contains( "*" ) || value.contains( "?" ))
```

Comment:
Move logic til util method with Javadoc to make code more expressive.

Reply:
done